plugins/scaffolder-backend/src/scaffolder/tasks/DatabaseTaskStore.ts

```diff
@@ -268,8 +269,7 @@ export class DatabaseTaskStore implements TaskStore {
         '<=',
         this.db.client.config.client.includes('sqlite3')
           ? this.db.raw(`datetime('now', ?)`, [`-${timeoutS} seconds`])
-          : this.db.raw(`dateadd('second', ?, ?)`, [
-              `-${timeoutS}`,
+          : this.db.raw(`? - interval '${timeoutS} seconds'`, [
```

Comment:
PostgressSQL DB was throwing that `dateadd` is not a function 😕 
so I had to use the `+/- interval` approach to add time to a date

Reply:
Just to chime in a little bit here, would like to dig in a little bit as to why `dateadd` doesn't work for you. Is it something to do with the PostgreSQL version? this works for most people right now, so it's unfortunate that it's not working for you. I mean the replacement passes the tests too, so it's probably fine, but I need some closure :joy:

Reply:
Sure, 
so here's what I get when I try to run the function with a PG Client:
![image](https://user-images.githubusercontent.com/18434722/189943798-6212e199-1b78-4d7f-b19c-d7cd711ccbab.png)

Same issue was being thrown when running `listStaleTasks`. 

I am using `latest` PG container, which is `v14.5`  according to their [Docker hub page](https://hub.docker.com/_/postgres).

On the other side, I couldn't find a `dateadd` operator in PG docs for "functions-datetime": 
* https://www.postgresql.org/docs/current/functions-datetime.html

Something that I noticed about `listStaleTasks` is that it was only being called by `TaskBroker.vacuumTasks` which is not being used acros Backstage code 🤔

Reply:
tbh I think it might be that the existing code is broken. Afaik postgres doesn't have a `dateadd` function in any version. It seems like the only DB tests that cover this code only runs for SQLite. Listing stale tasks isn't something that's ever done until now, so it's entirely possible that it simply doesn't work atm.

tl;dr: :shipit: